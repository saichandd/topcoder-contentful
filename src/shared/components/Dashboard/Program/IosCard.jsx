import React from 'react';
import PT from 'prop-types';

import { stripUnderscore } from 'utils/tc';
import { config } from 'topcoder-react-utils';

import './IosCard.scss';

const IosCard = (props) => {
  const { challenge } = props;
  return (
    <div styleName={`challenge tile-view ${challenge.track}`}>
      <div styleName="challenge-track" />
      <header>
        <div>
          <a href={`${config.URL.BASE}/challenge-details/${challenge.id}/?type=${challenge.track}`} styleName="name">
            <span>
              {challenge.name}
            </span>
          </a>
          <p styleName="subtrack-color">
            {stripUnderscore(challenge.subTrack)}
          </p>
        </div>
        <div styleName="challenge-links">
          <div styleName="stats">
            <a href={`${config.URL.BASE}/challenge-details/${challenge.id}/?type=${challenge.track}#viewRegistrant`} styleName="registrants">
              <div styleName="registrants-icon" />
              <p>
                {challenge.numRegistrants}
              </p>
            </a>
            <a href={`${config.URL.BASE}/challenge-details/${challenge.id}/?type=${challenge.track}#submissions`} styleName="submissions">
              <div styleName="submissions-icon" />
              <p>
                {challenge.numSubmissions}
              </p>
            </a>
          </div>
          <a href={`${config.URL.FORUMS}/?module=Category&categoryID=${challenge.forumId}`} styleName="forum">
            <div styleName="forum-icon" />
            <p>
Posts
            </p>
          </a>
        </div>
      </header>
      <div styleName="challenge-details">
        <p styleName="currentPhase">
          {challenge.userCurrentPhase}
        </p>
        {
          challenge.userCurrentPhaseEndTime
          && (
          <div styleName="challenge-calendar">
            <p>
Ends In
            </p>
            <p styleName="time-remaining">
              {challenge.userCurrentPhaseEndTime[0]}
            </p>
            <p styleName="unit-of-time">
              {challenge.userCurrentPhaseEndTime[1]}
            </p>
          </div>
          )
        }
        {
          !challenge.userCurrentPhaseEndTime
          && (
          <div styleName="stalled-challenge">
This challenge is currently paused.
          </div>
          )
        }
        {
          challenge.reviewType === 'PEER'
          && (
          <p styleName="prize-money">
Peer Review Challenge
          </p>
          )
        }
        {
          challenge.reviewType !== 'PEER'
          && (
          <p styleName="prize-money">
            {`$${(challenge.totalPrize || 0).toLocaleString()}`}
          </p>
          )
        }
        <p styleName="technologies">
          {challenge.technologies}
        </p>
      </div>
    </div>
  );
};

IosCard.propTypes = {
  challenge: PT.shape().isRequired,
};

export default IosCard;
