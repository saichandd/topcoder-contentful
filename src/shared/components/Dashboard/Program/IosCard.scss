@import '~styles/mixins';

// Default Challenge Tile Stylings
.challenge.tile-view {
  position: relative;
  width: 270px;
  height: 370px;
  border: 1px solid #dcdcdc;
  border-radius: 4px;
  background-color: $tc-white;

  .challenge-track {
    width: 5px;
    height: 91px;
    position: absolute;
    top: -1px;
    left: -1px;
    border-top-left-radius: 4px;
  }

  .challenge-calendar {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 75px;
    height: 63px;
    margin-top: 16px;
    background-image: url(assets/images/dashboard/ico-calendar.svg);

    > p {
      font-family: 'Sofia Pro', Arial, Helvetica, sans-serif;
      font-weight: bold;
    }

    .starts-in {
      margin-top: 3px;
      font-size: 10px;
      line-height: 13px;
      text-transform: uppercase;
      color: #7f7f7f;
    }

    .time-remaining {
      margin-top: 5px;
      font-size: 24px;
      color: #3d3d3d;
    }

    .unit-of-time {
      margin-top: 1px;
      font-size: 10px;
      text-transform: lowercase;
      color: #7f7f7f;
    }
  }

  header {
    height: 91px;
    border-bottom: 1px solid #f0f0f0;
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    a.name {
      display: block;
      padding: 15px 20px 0;
      font-family: 'Sofia Pro', Arial, Helvetica, sans-serif;
      font-weight: 500;
      font-size: 14px;
      line-height: 20px;
      color: #3d3d3d;
      text-decoration: none;

      @include ellipsis;

      text-transform: uppercase;
    }

    p.subtrack-color {
      padding: 0 20px;
      margin-top: 5px;
      font-family: 'Sofia Pro', Arial, Helvetica, sans-serif;
      font-weight: 500;
      font-size: 12px;
      line-height: 14px;
      text-transform: uppercase;
    }

    .challenge-links {
      display: flex;
      flex-direction: row;
      align-items: center;
      justify-content: space-between;
      width: 100%;
      border-bottom-right-radius: 3px;
      border-bottom-left-radius: 3px;
      padding: 0 20px;
      margin-bottom: 10px;

      a {
        text-decoration: none;

        p {
          color: #a3a3ae;
          font-family: 'Sofia Pro', Arial, Helvetica, sans-serif;
          font-weight: 500;
          font-size: 12px;
          line-height: 14px;
          text-transform: uppercase;
        }
      }

      .stats,
      .registrants,
      .submissions,
      .forum {
        display: flex;
        flex-direction: row;
        align-items: center;
      }

      .registrants,
      .submissions,
      .form {
        cursor: pointer;
      }

      .registrants {
        margin-right: 15px;
      }
    }
  }

  .challenge-details {
    display: flex;
    flex-direction: column;
    align-items: center;

    .currentPhase {
      margin-top: 40px;
      font-family: 'Sofia Pro', Arial, Helvetica, sans-serif;
      font-weight: 300;
      font-size: 18px;
      line-height: 23px;
      text-transform: uppercase;
      color: #3d3d3d;
    }
  }

  .prize-money {
    margin-top: 32px;
    margin-bottom: 10px;
    font-family: 'Sofia Pro', Arial, Helvetica, sans-serif;
    font-weight: 300;
    font-size: 20px;
    line-height: 24px;
  }

  .technologies {
    height: 36px;
    margin-bottom: 33px;
    padding: 0 20px;
    font-family: 'Merriweather Sans', Arial, Helvetica, sans-serif;
    font-size: 13px;
    line-height: 18px;
  }

  .stalled-challenge {
    margin-top: 31px;
    margin-bottom: 12px;
    height: 36px;
    padding: 0 20px;
    font-family: 'Sofia Pro', Arial, Helvetica, sans-serif;
    font-weight: 500;
    font-size: 12px;
    line-height: 18px;
    text-transform: uppercase;
    text-align: center;
    color: #a3a3ae;
  }

  .phase-action {
    margin-bottom: 25px;
  }

  .roles {
    min-height: 36px;
    margin-bottom: 25px;
    color: #3d3d3d;
    font-family: 'Merriweather Sans', Arial, Helvetica, sans-serif;
    font-size: 13px;
    line-height: 18px;
  }

  div[class$="-icon"] {
    margin-right: 3px;
  }

  .registrants-icon {
    width: 15px;
    height: 16px;
    background-image: url(assets/images/dashboard/ico-users.svg);
    background-size: 15px 16px;
  }

  .submissions-icon {
    width: 22px;
    height: 14px;
    background-image: url(assets/images/dashboard/ico-submissions.svg);
  }

  .forum-icon {
    width: 20px;
    height: 17px;
    background-image: url(assets/images/dashboard/ico-posts.svg);
  }

  // Dynamic colors based on track
  &.DESIGN {
    .challenge-track {
      background-color: #21b2f1;
    }

    header p.subtrack-color {
      color: #21b2f1;
    }
  }

  &.DEVELOP {
    .challenge-track {
      background-color: #05c14f;
    }

    header p.subtrack-color {
      color: #05c14f;
    }
  }

  &.DATA_SCIENCE {
    .challenge-track {
      background-color: #fc9a00;
    }

    header p.subtrack-color {
      color: #fc9a00;
    }
  }

  &.COPILOT {
    .challenge-track {
      background-color: #7d939e;
    }

    header p.subtrack-color {
      color: #7d939e;
    }
  }
}
