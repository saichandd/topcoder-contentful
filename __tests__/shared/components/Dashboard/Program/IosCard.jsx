/* global window */

import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import IosCard from 'components/Dashboard/Program/IosCard';

const mockData1 = {
  challenge: {
    id: '1',
    name: 'name',
  },
};

const mockData2 = {
  challenge: {
    id: '1',
    name: 'name',
    userCurrentPhaseEndTime: [1, 2],
  },
};

const mockData3 = {
  challenge: {
    id: '1',
    name: 'name',
    reviewType: 'PEER',
  },
};

test.skip('Matches shallow shapshot', () => {
  const renderer = new Renderer();

  renderer.render((
    <IosCard {...mockData1} />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();

  renderer.render((
    <IosCard {...mockData2} />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();

  Object.defineProperty(window.location, 'href', {
    writable: true,
    value: 'http://members.topcoder.com',
  });
  renderer.render((
    <IosCard {...mockData3} />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
