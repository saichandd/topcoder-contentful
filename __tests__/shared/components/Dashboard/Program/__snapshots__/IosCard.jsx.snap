// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<div
  className="src-shared-components-Dashboard-Program-___IosCard__challenge___3ZUuF src-shared-components-Dashboard-Program-___IosCard__tile-view___1rHjP"
>
  <div
    className="src-shared-components-Dashboard-Program-___IosCard__challenge-track___h63e3"
  />
  <header>
    <div>
      <a
        className="src-shared-components-Dashboard-Program-___IosCard__name___TOMHJ"
        href="https://www.topcoder-dev.com/challenge-details/1/?type=undefined"
      >
        <span>
          name
        </span>
      </a>
      <p
        className="src-shared-components-Dashboard-Program-___IosCard__subtrack-color___2axPS"
      >
        
      </p>
    </div>
    <div
      className="src-shared-components-Dashboard-Program-___IosCard__challenge-links___2nxNN"
    >
      <div
        className="src-shared-components-Dashboard-Program-___IosCard__stats___1AMfH"
      >
        <a
          className="src-shared-components-Dashboard-Program-___IosCard__registrants___gN_NW"
          href="https://www.topcoder-dev.com/challenge-details/1/?type=undefined#viewRegistrant"
        >
          <div
            className="src-shared-components-Dashboard-Program-___IosCard__registrants-icon___2PTVj"
          />
          <p />
        </a>
        <a
          className="src-shared-components-Dashboard-Program-___IosCard__submissions___2AfFz"
          href="https://www.topcoder-dev.com/challenge-details/1/?type=undefined#submissions"
        >
          <div
            className="src-shared-components-Dashboard-Program-___IosCard__submissions-icon___2xhE2"
          />
          <p />
        </a>
      </div>
      <a
        className="src-shared-components-Dashboard-Program-___IosCard__forum___68F-9"
        href="https://apps.topcoder-dev.com/forums/?module=Category&categoryID=undefined"
      >
        <div
          className="src-shared-components-Dashboard-Program-___IosCard__forum-icon___3_Xe5"
        />
        <p>
          Posts
        </p>
      </a>
    </div>
  </header>
  <div
    className="src-shared-components-Dashboard-Program-___IosCard__challenge-details___26M3D"
  >
    <p
      className="src-shared-components-Dashboard-Program-___IosCard__currentPhase___1ebbi"
    />
    <div
      className="src-shared-components-Dashboard-Program-___IosCard__stalled-challenge___2EDQf"
    >
      This challenge is currently paused.
    </div>
    <p
      className="src-shared-components-Dashboard-Program-___IosCard__prize-money___sw4lK"
    >
      $0
    </p>
    <p
      className="src-shared-components-Dashboard-Program-___IosCard__technologies___YLHCt"
    />
  </div>
</div>
`;

exports[`Matches shallow shapshot 2`] = `
<div
  className="src-shared-components-Dashboard-Program-___IosCard__challenge___3ZUuF src-shared-components-Dashboard-Program-___IosCard__tile-view___1rHjP"
>
  <div
    className="src-shared-components-Dashboard-Program-___IosCard__challenge-track___h63e3"
  />
  <header>
    <div>
      <a
        className="src-shared-components-Dashboard-Program-___IosCard__name___TOMHJ"
        href="https://www.topcoder-dev.com/challenge-details/1/?type=undefined"
      >
        <span>
          name
        </span>
      </a>
      <p
        className="src-shared-components-Dashboard-Program-___IosCard__subtrack-color___2axPS"
      >
        
      </p>
    </div>
    <div
      className="src-shared-components-Dashboard-Program-___IosCard__challenge-links___2nxNN"
    >
      <div
        className="src-shared-components-Dashboard-Program-___IosCard__stats___1AMfH"
      >
        <a
          className="src-shared-components-Dashboard-Program-___IosCard__registrants___gN_NW"
          href="https://www.topcoder-dev.com/challenge-details/1/?type=undefined#viewRegistrant"
        >
          <div
            className="src-shared-components-Dashboard-Program-___IosCard__registrants-icon___2PTVj"
          />
          <p />
        </a>
        <a
          className="src-shared-components-Dashboard-Program-___IosCard__submissions___2AfFz"
          href="https://www.topcoder-dev.com/challenge-details/1/?type=undefined#submissions"
        >
          <div
            className="src-shared-components-Dashboard-Program-___IosCard__submissions-icon___2xhE2"
          />
          <p />
        </a>
      </div>
      <a
        className="src-shared-components-Dashboard-Program-___IosCard__forum___68F-9"
        href="https://apps.topcoder-dev.com/forums/?module=Category&categoryID=undefined"
      >
        <div
          className="src-shared-components-Dashboard-Program-___IosCard__forum-icon___3_Xe5"
        />
        <p>
          Posts
        </p>
      </a>
    </div>
  </header>
  <div
    className="src-shared-components-Dashboard-Program-___IosCard__challenge-details___26M3D"
  >
    <p
      className="src-shared-components-Dashboard-Program-___IosCard__currentPhase___1ebbi"
    />
    <div
      className="src-shared-components-Dashboard-Program-___IosCard__challenge-calendar___14oal"
    >
      <p>
        Ends In
      </p>
      <p
        className="src-shared-components-Dashboard-Program-___IosCard__time-remaining___1YYsW"
      >
        1
      </p>
      <p
        className="src-shared-components-Dashboard-Program-___IosCard__unit-of-time___Tfjle"
      >
        2
      </p>
    </div>
    <p
      className="src-shared-components-Dashboard-Program-___IosCard__prize-money___sw4lK"
    >
      $0
    </p>
    <p
      className="src-shared-components-Dashboard-Program-___IosCard__technologies___YLHCt"
    />
  </div>
</div>
`;

exports[`Matches shallow shapshot 3`] = `
<div
  className="src-shared-components-Dashboard-Program-___IosCard__challenge___3ZUuF src-shared-components-Dashboard-Program-___IosCard__tile-view___1rHjP"
>
  <div
    className="src-shared-components-Dashboard-Program-___IosCard__challenge-track___h63e3"
  />
  <header>
    <div>
      <a
        className="src-shared-components-Dashboard-Program-___IosCard__name___TOMHJ"
        href="https://www.topcoder-dev.com/challenge-details/1/?type=undefined"
      >
        <span>
          name
        </span>
      </a>
      <p
        className="src-shared-components-Dashboard-Program-___IosCard__subtrack-color___2axPS"
      >
        
      </p>
    </div>
    <div
      className="src-shared-components-Dashboard-Program-___IosCard__challenge-links___2nxNN"
    >
      <div
        className="src-shared-components-Dashboard-Program-___IosCard__stats___1AMfH"
      >
        <a
          className="src-shared-components-Dashboard-Program-___IosCard__registrants___gN_NW"
          href="https://www.topcoder-dev.com/challenge-details/1/?type=undefined#viewRegistrant"
        >
          <div
            className="src-shared-components-Dashboard-Program-___IosCard__registrants-icon___2PTVj"
          />
          <p />
        </a>
        <a
          className="src-shared-components-Dashboard-Program-___IosCard__submissions___2AfFz"
          href="https://www.topcoder-dev.com/challenge-details/1/?type=undefined#submissions"
        >
          <div
            className="src-shared-components-Dashboard-Program-___IosCard__submissions-icon___2xhE2"
          />
          <p />
        </a>
      </div>
      <a
        className="src-shared-components-Dashboard-Program-___IosCard__forum___68F-9"
        href="https://apps.topcoder-dev.com/forums/?module=Category&categoryID=undefined"
      >
        <div
          className="src-shared-components-Dashboard-Program-___IosCard__forum-icon___3_Xe5"
        />
        <p>
          Posts
        </p>
      </a>
    </div>
  </header>
  <div
    className="src-shared-components-Dashboard-Program-___IosCard__challenge-details___26M3D"
  >
    <p
      className="src-shared-components-Dashboard-Program-___IosCard__currentPhase___1ebbi"
    />
    <div
      className="src-shared-components-Dashboard-Program-___IosCard__stalled-challenge___2EDQf"
    >
      This challenge is currently paused.
    </div>
    <p
      className="src-shared-components-Dashboard-Program-___IosCard__prize-money___sw4lK"
    >
      Peer Review Challenge
    </p>
    <p
      className="src-shared-components-Dashboard-Program-___IosCard__technologies___YLHCt"
    />
  </div>
</div>
`;
